### ISSUE TITLE
sql: ARRAYs only containing nulls don't work properly

### ISSUE THREAD

**Issue by justinj** [cockroachdb/cockroach#29355]
```
SELECT bar FROM foo WHERE bar ?| ARRAY[null, null]::STRING[]
```

The type doesn't get propagated properly because we don't serialize the array with the proper type annotation. I spent some time going down that rabbit hole before giving up. It's not impossible, just will take some time.

**Comment by jordanlewis** [comment]
cc @mjibson, this seems related to the other nulls-in-arrays issue we were looking at.

**Comment by knz** [comment]
This is not a S-1 severity.

**Comment by jordanlewis** [comment]
This was fixed by #36673.
